### DISCUSSION TITLE
erase json object from a file

### DISCUSSION THREAD

**Discussion by barweizman** [danielaparker/jsoncons#289] [Q&A]
I got a JSON file (written below), and I want to **erase the JSON object with the given id** "756746783" (this is the first object in the example), **with all its self-other attributes**(e.g address, email..), How can I do that?

[
    {
        "address": "ashdod", 
        "email": "ron10@gmail.com", 
        "emergency contact name": "alon", 
        "emergency contact phone": "0535500127", 
        "first name": "ron", 
        "id": "756746783", 
        "last name": "cohen", 
        "password": "77ik2331", 
        "phone": "0526732996", 
        "reset password details": ["olga", "06.11.1995", "tennis"], 
        "salary": 3000, 
        "type": "manager", 
        "user name": "ron87"
    }, 
    {
        "address": "ashdod", 
        "email": "nirlevy120@gmail.com", 
        "emergency contact name": "tal", 
        "emergency contact phone": "0548177765", 
        "first name": "nir", 
        "id": "11884398", 
        "last name": "levy", 
        "password": "68fnsd4", 
        "phone": "0578198932", 
        "reset password details": ["koral", "02.01.1989", "sleep"], 
        "salary": 4500, 
        "type": "manager", 
        "user name": "nir97"
    }
]

**Comment by danielaparker** [comment]
Here's one way:

```
#include <jsoncons/json.hpp>
#include <fstream>

using jsoncons::json;

int main()
{
    try
    {
        // Read from input file
        std::ifstream is(/* pathname_to_input */);
 
        // If you want to preserve the order of members to be the same 
        // as in the original file, use ojson instead,
        json instance = json::parse(is);
 
        // Locate the item to be erased
        auto it = instance.array_range().begin();
        while (it != instance.array_range().end())
        {
            if (it->at("id").as<std::string>() == "756746783")
            {
                break;
            }
        }
 
        // If found, erase it
        if (it != instance.array_range().end())
        {
            instance.erase(it);
        }

        // Write to output file
        std::ofstream os(/* pathname_to_output */);

        instance.dump_pretty(os);
    }
    catch (const std::exception& e)
    {
        std::cout << e.what() << std::endl;    
    }
}
```

Here's another way using jsonpath:

```
#include <iostream>
#include <fstream>
#include <jsoncons/json.hpp>
#include <jsoncons_ext/jsonpath/jsonpath.hpp>

using jsoncons::ojson;
namespace jsonpath = jsoncons::jsonpath;

int main()
{
    std::ifstream is(/*path/to/input*/);

    ojson instance = ojson::parse(is);

    // Select all objects that have id not equal to '756746783' and return in an array 
    auto result = jsonpath::json_query(instance, "$.*[?(@.id != '756746783')]");

    std::cout << pretty_print(result) << "\n";

    std::ofstream os(/*path/to/output*/);
    result.dump_pretty(os);
}
```

**Reply by barweizman** [reply]
Definitely what I was looking for! This is my first time using json objects and working with json files, I searched everywhere and couldn`t find an appropriate example. I really recommend you to add this to the examples folder, very helpful. 
Thank you!

**Comment by danielaparker** [comment]
It's been added [Erase an object with a specified key from an array](https://github.com/danielaparker/jsoncons/blob/master/doc/Examples.md#J3), and the examples folder. Thanks for the contribution.
